### PR TITLE
refactor: [M3-7893] – Post-GA VPC cleanup

### PR DESCRIPTION
**Pull request by dwiley-akamai**
## Description 📝
Since VPC entered GA several weeks back, we are now able to circle back to complete some code cleanup. This ticket is primarily for `@TODO VPC` items related to the VPC feature flag and related conditionals.

## Changes  🔄
- Clean up `flags.vpc`, `showVPCs`, and related logic throughout many files
- Remove `manuallyUpdateConfigs` prop from `PowerActionsDialogOrDrawer.tsx`
- Use `import type { ...}` convention in a few places

## Target release date 🗓️
04/01/2024

### Verification steps
Ensure there have been no regressions compared to prod, specifically:

- VPC appears in Create button dropdown and in PrimaryNav
- VPC landing and detail pages load properly
- VPC appears in list when you hit `Ctrl` + `Shift` + `K`
- For Linodes assigned to a VPC...
  - "VPC" section is present in the header panel on the Linode Details page
- For Linode Create flow...
  - VLAN accordion is present
  - `"Use Private IP for a backend node to a NodeBalancer. Use VPC instead for private communication between your Linodes."` copy is present for Private IP add-on
- Powering Linodes on and off
- `InterfaceSelect` works as expected in Linode Config dialog and VLAN accordion
- "VPC" appears in "Add Personal Access Token" drawer & "View Scopes" drawer for PATs
- No adverse impact to VPC Subnets table

## As an Author I have considered 🤔
- [X] 👀 Doing a self review
- [X] ❔ Our [contribution guidelines](https://github.com/linode/manager/blob/develop/docs/CONTRIBUTING.md)
- [X] 🤏 Splitting feature into small PRs
- [X] ➕ Adding a changeset
- [X] 🧪 Providing/Improving test coverage
- [X] 🔐 Removing all sensitive information from the code and PR description
- [X] 🚩 Using a feature flag to protect the release
- [X] 👣 Providing comprehensive reproduction steps
- [X] 📑 Providing or updating our documentation
- [X] 🕛 Scheduling a pair reviewing session
- [X] 📱 Providing mobile support
- [X] ♿  Providing accessibility support

### CODE REVIEW

**Comment by dwiley-akamai** [comment]
The removal of VPC from `featureFlags.ts` and `FeatureFlagTool.tsx` will be done in M3-7895

File: packages/manager/src/features/Linodes/LinodesCreate/VPCPanel.tsx [hunk 1]
```diff
@@ -95,29 +91,14 @@ export const VPCPanel = (props: VPCPanelProps) => {
...
-    account?.capabilities ?? []
-  );
-
-  // To prevent unnecessary API calls, pass the displayVPCPanel boolean to determine whether the query is enabled
-  const { data: vpcData, error, isLoading } = useVPCsQuery(
-    {},
-    {},
-    displayVPCPanel
-  );
+  const { data: vpcData, error, isLoading } = useVPCsQuery({}, {});
```
Comment (bnussman-akamai) [inline packages/manager/src/features/Linodes/LinodesCreate/VPCPanel.tsx:94]: If the uses has more the 100 VPCs, only the first 100 will show

**Review by bnussman-akamai — APPROVED** [review]
Great clean up!

File: packages/manager/src/features/Linodes/LinodesCreate/VPCPanel.tsx (hunk 1 above)
Comment (hana-linode) [reply packages/manager/src/features/Linodes/LinodesCreate/VPCPanel.tsx:94]: @dwiley-akamai We can probably remove the `enabled` argument for `useVPCsQuery` now

**Review by hana-linode — APPROVED** [review]
Found a minor existing bug with the event progress bar width but otherwise LGTM! 🧹 

image

**Comment by dwiley-akamai** [comment]
> Found a minor existing bug with the event progress bar width but otherwise LGTM! 🧹
> 
> image

This is when powering down a linode, right?

File: packages/manager/src/features/Linodes/LinodesCreate/VPCPanel.tsx (hunk 1 above)
Comment (dwiley-akamai) [reply packages/manager/src/features/Linodes/LinodesCreate/VPCPanel.tsx:94]: Going to create a ticket for implementing a `useAllVPCsQuery` where appropriate

**Pull request merged**